### PR TITLE
Technical: changelog-tag-based. New logic for development, release, sos, master branches

### PR DESCRIPTION
**Pull request by mychkin**

### CODE REVIEW

File: changelog-extractor.py [hunk 1]
```diff
@@ -125,19 +125,42 @@ def send_message_to_slack(text, channel, token, username) :
...
+    # development branch
+    if not there_is_a_tag:
+        print "development case"
+        old_tag = run_command("git describe --abbrev=0 --match v*-rc ")
+   
+    # release case
+    if there_is_a_release_tag:
+        print "release case"
+        new_tag = os.environ["CI_BUILD_TAG"]
+        old_tag = run_command("git describe --abbrev=0 --tags "+ new_tag +"^ --match v*[0-9]-rc")
```
Comment (NunoPinheiro) [inline changelog-extractor.py:142]: Is this regex right?
This is the one used by other scripts:
v([\d]+\.){2}([\d]+)(-rc)

Your regex seems to work for v11-rc but not for v11.11.11-rc which is the format we are using

Comment (mychkin) [reply changelog-extractor.py:142]: it actually works for v11.11.11-rc

try to execute command 
git describe --abbrev=0 --match v*[0-9]-rc
git describe --abbrev=0 --match v*[0-9]
git describe --abbrev=0 --match v*[0-9]-sos

Comment (mychkin) [reply changelog-extractor.py:142]: In order to test "release case" 
1) add a tag with -rc suffix
2) run command git describe --abbrev=0 --tags v6.40.1-rc^ --match v*[0-9]-rc

Comment (mychkin) [reply changelog-extractor.py:142]: I was trying to use expression from other scripts, but it looks like we need to escape some characters in v([\d]+.){2}([\d]+)(-rc) in order to make it work in --match

**Comment by mychkin** [comment]
btw, the problem with this solution is that it does not order tags (it uses default order by "createdAt" date). In other words, if someone create tag v6.1.1-rc and after that v6.1.0-rc the script will consider tag  v6.1.0-rc as "younger tag"

**Review by NunoPinheiro — APPROVED** [review]

**Pull request merged**